File: VKRecyclerViewFragment/app/src/main/java/com/example/vkrecyclerview/NewsDetailActivity.java
package com.example.vkrecyclerview;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.os.Parcelable;
import android.util.SparseArray;
import android.view.Gravity;
import android.view.View;
import android.widget.ImageView;
import android.widget.ScrollView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;

import static android.view.View.NO_ID;

public class NewsDetailActivity extends AppCompatActivity {

    ImageView accountPhotoUrl;
    TextView accountName;
    TextView date;
    TextView newsText;
    ImageView postPhotoUrl;
    ImageView likesUrl;
    TextView likes;
    ImageView userLike1;
    ImageView userLike2;
    TextView likesDetail;
    TextView showCom;//private String showCom;
    ImageView userComUrl1;//private int userComUrl1;
    TextView userComAcc1;//private String userComAcc1;
    TextView userComText1;//private String userComText1;
    TextView userComDate1;//private String userComDate1;
    ImageView userComReplyUrl1;//private int userComReplyUrl1;
    ImageView userComLikeUrl1;//private int userComLikeUrl1;
    TextView userComLike1;//private int userComLike1;
    ImageView postsUrl;
    TextView postsCount;
    ImageView viewersUrl;
    TextView viewersCount;
    private int isLiked;
    private int likesCount;



    public void onBackPressed() {
        Intent intent = new Intent(NewsDetailActivity.this, MainActivity.class);
        intent.putExtra("pagerState", 1);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
        overridePendingTransition(0, 0);

    }



    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_news_detail);
        accountPhotoUrl = findViewById(R.id.accountPhotoUrl);
        accountName = findViewById(R.id.accountName);
        date = findViewById(R.id.date);
        newsText = findViewById(R.id.newsText);
        postPhotoUrl = findViewById(R.id.postPhotoUrl);
        likesUrl = findViewById(R.id.likesUrl);
        likes = findViewById(R.id.likesCount);
        userLike1=findViewById(R.id.userLike1);
        userLike2=findViewById(R.id.userLike2);
        likesDetail=findViewById(R.id.likesDetail);
        showCom=findViewById(R.id.showCom);//private String showCom;
        userComUrl1=findViewById(R.id.userComUrl1);//private int userComUrl1;
        userComAcc1=findViewById(R.id.userComAcc1);//private String userComAcc1;
        userComText1=findViewById(R.id.userComText1);//private String userComText1;
        userComDate1=findViewById(R.id.userComDate1);//private String userComDate1;
        userComReplyUrl1=findViewById(R.id.userComReplyUrl1);//private int userComReplyUrl1;
        userComLikeUrl1=findViewById(R.id.userComLikeUrl1);//private int userComLikeUrl1;
        userComLike1=findViewById(R.id.userComLike1);//private int userComLike1;
        postsUrl = findViewById(R.id.postsUrl);
        postsCount = findViewById(R.id.postsCount);
        viewersUrl = findViewById(R.id.viewersUrl);
        viewersCount = findViewById(R.id.viewersCount);

        final News news = (News) getIntent().getParcelableExtra("news");

        likesCount = news.getLikesCount();

        if(news.getLiked()==1){
            Glide.with(likesUrl.getContext())
                    .load(R.drawable.likeblue)
                    .into(likesUrl);
            likes.setTextColor(Color.parseColor("#006ba8"));
        }
        else{
            Glide.with(likesUrl.getContext())
                    .load(R.drawable.likemin)
                    .into(likesUrl);
            likes.setTextColor(Color.parseColor("#979899"));
        }


        Glide.with(accountPhotoUrl.getContext())
                .load(news.getAccountPhotoUrl())
                .into(accountPhotoUrl);

        accountName.setText(news.getAccountName());

        date.setText(news.getDate());

        newsText.setText(news.getNewsText());

        Glide.with(postPhotoUrl.getContext())
                .load(news.getPostPhotoUrl())
                .into(postPhotoUrl);

        likes.setText(Integer.toString(news.getLikesCount()));

        Glide.with(userLike1.getContext())
                .load(news.getUserLike1())
                .into(userLike1);

        Glide.with(userLike2.getContext())
                .load(news.getUserLike2())
                .into(userLike2);

        likesDetail.setText(news.getLikesDetail());

        showCom.setText(news.getShowCom());

        Glide.with(userComUrl1.getContext())
                .load(news.getUserComUrl1())
                .into(userComUrl1);

        userComAcc1.setText(news.getUserComAcc1());

        userComText1.setText(news.getUserComText1());

        userComDate1.setText(news.getUserComDate1());

        Glide.with(userComReplyUrl1.getContext())
                .load(news.getUserComReplyUrl1())
                .into(userComReplyUrl1);

        Glide.with(userComLikeUrl1.getContext())
                .load(news.getUserComLikeUrl1())
                .into(userComLikeUrl1);

        userComLike1.setText(Integer.toString(news.getUserComLike1()));

        Glide.with(postsUrl.getContext())
                .load(news.getPostsUrl())
                .into(postsUrl);

        Glide.with(postsUrl.getContext())
                .load(news.getPostsUrl())
                .into(postsUrl);

        postsCount.setText(String.valueOf(news.getPostsCount()));

        Glide.with(viewersUrl.getContext())
                .load(news.getViewersUrl())
                .into(viewersUrl);

        viewersCount.setText(String.valueOf(news.getViewersCount()));

        likesUrl.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(news.getLiked()==1){
                    if(SecondFragment.favoriteItems.contains(news)){
                        SecondFragment.favoriteItems.remove(news);
                    }
                    for(News item:DataBase.news){
                        if(item.equals(news)){
                            item.setLiked(0);
                            item.setLikesCount(item.getLikesCount()-1);
                        }
                    }
                    news.setLiked(0);
                    Glide.with(likesUrl.getContext())
                            .load(R.drawable.likemin)
                            .into(likesUrl);
                    news.setLikesCount(news.getLikesCount()-1);
                    likes.setText(Integer.toString(news.getLikesCount()));
                    likes.setTextColor(Color.parseColor("#979899"));
                    Toast toast = Toast.makeText(v.getContext(),
                            "Like removed!",
                            Toast.LENGTH_SHORT);
                    toast.setGravity(Gravity.CENTER, 0, 0);
                    toast.show();
                }
                else{
                    for (News item:DataBase.news){
                        if(item.equals(news)){
                            item.setLiked(1);
                            item.setLikesCount(item.getLikesCount()+1);
                        }
                    }
                    news.setLiked(1);
                    Glide.with(likesUrl.getContext())
                            .load(R.drawable.likebluemin)
                            .into(likesUrl);

                    news.setLikesCount(news.getLikesCount()+1);

                    likes.setText(Integer.toString(news.getLikesCount()));
                    likes.setTextColor(Color.parseColor("#006ba8"));
                    if(!SecondFragment.favoriteItems.contains(news)) {
                        SecondFragment.favoriteItems.add(news);
                    }
                    Toast toast = Toast.makeText(v.getContext(),
                            "Liked!",
                            Toast.LENGTH_SHORT);
                    toast.setGravity(Gravity.CENTER, 0, 0);
                    toast.show();
                }
            }
        });
    }

}

File: app/src/main/java/com/example/vkrecyclerviewkotlin/NewsDetailActivity.kt
package com.example.vkrecyclerviewkotlin

import android.annotation.SuppressLint
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.os.Parcelable
import android.view.Gravity
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide

class NewsDetailActivity : AppCompatActivity() {
    private lateinit var accountPhotoUrl: ImageView
    private lateinit var accountName: TextView
    private lateinit var date: TextView
    private lateinit var newsText: TextView
    private lateinit var postPhotoUrl: ImageView
    private lateinit var likesUrl: ImageView
    private lateinit var likes: TextView
    private lateinit var userLike1: ImageView
    private lateinit var userLike2: ImageView
    private lateinit var likesDetail: TextView
    private lateinit var showCom: TextView
    private lateinit var userComUrl1: ImageView
    private lateinit var userComAcc1: TextView
    private lateinit var userComText1: TextView
    private lateinit var userComDate1: TextView
    private lateinit var userComReplyUrl1: ImageView
    private lateinit var userComLikeUrl1: ImageView
    private lateinit var userComLike1: TextView
    private lateinit var postsUrl: ImageView
    private lateinit var postsCount: TextView
    private lateinit var viewersUrl: ImageView
    private lateinit var viewersCount: TextView
    private var likesCount = 0
    private lateinit var thisNews: News

    override fun onBackPressed() {
        DataBase.news[thisNews.id] = thisNews
        val intent = Intent(this@NewsDetailActivity, MainActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
        startActivity(intent)
    }

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_news_detail)
        accountPhotoUrl = findViewById(R.id.accountPhotoUrl)
        accountName = findViewById(R.id.accountName)
        date = findViewById(R.id.date)
        newsText = findViewById(R.id.newsText)
        postPhotoUrl = findViewById(R.id.postPhotoUrl)
        likesUrl = findViewById(R.id.likesUrl)
        likes = findViewById(R.id.likesCount)
        userLike1 = findViewById(R.id.userLike1)
        userLike2 = findViewById(R.id.userLike2)
        likesDetail = findViewById(R.id.likesDetail)
        showCom = findViewById(R.id.showCom) //private String showCom;
        userComUrl1 =
            findViewById(R.id.userComUrl1) //private int userComUrl1;
        userComAcc1 = findViewById(R.id.userComAcc1) //private String userComAcc1;
        userComText1 = findViewById(R.id.userComText1) //private String userComText1;
        userComDate1 = findViewById(R.id.userComDate1) //private String userComDate1;
        userComReplyUrl1 =
            findViewById(R.id.userComReplyUrl1) //private int userComReplyUrl1;
        userComLikeUrl1 =
            findViewById(R.id.userComLikeUrl1) //private int userComLikeUrl1;
        userComLike1 = findViewById(R.id.userComLike1) //private int userComLike1;
        postsUrl = findViewById(R.id.postsUrl)
        postsCount = findViewById(R.id.postsCount)
        viewersUrl = findViewById(R.id.viewersUrl)
        viewersCount = findViewById(R.id.viewersCount)
        val news = intent.getParcelableExtra<Parcelable>("news") as News
        thisNews = news
        likesCount = news.likesCount
        if (news.liked == 1) {
            Glide.with(likesUrl.context)
                .load(R.drawable.likeblue)
                .into(likesUrl)
            likes.setTextColor(Color.parseColor("#006ba8"))
        } else {
            Glide.with(likesUrl.context)
                .load(R.drawable.likemin)
                .into(likesUrl)
            likes.setTextColor(Color.parseColor("#979899"))
        }
        Glide.with(accountPhotoUrl.context)
            .load(news.accountPhotoUrl)
            .into(accountPhotoUrl)
        accountName.text = news.accountName
        date.text = news.date
        newsText.text = news.newsText
        Glide.with(postPhotoUrl.context)
            .load(news.postPhotoUrl)
            .into(postPhotoUrl)
        likes.text = news.likesCount.toString()
        Glide.with(userLike1.context)
            .load(news.userLike1)
            .into(userLike1)
        Glide.with(userLike2.context)
            .load(news.userLike2)
            .into(userLike2)
        likesDetail.text = news.likesDetail
        showCom.text = news.showCom
        Glide.with(userComUrl1.context)
            .load(news.userComUrl1)
            .into(userComUrl1)
        userComAcc1.text = news.userComAcc1
        userComText1.text = news.userComText1
        userComDate1.text = news.userComDate1
        Glide.with(userComReplyUrl1.context)
            .load(news.userComReplyUrl1)
            .into(userComReplyUrl1)
        Glide.with(userComLikeUrl1.context)
            .load(news.userComLikeUrl1)
            .into(userComLikeUrl1)
        userComLike1.text = news.userComLike1.toString()
        Glide.with(postsUrl.context)
            .load(news.postsUrl)
            .into(postsUrl)
        Glide.with(postsUrl.context)
            .load(news.postsUrl)
            .into(postsUrl)
        postsCount.text = news.postsCount.toString()
        Glide.with(viewersUrl.context)
            .load(news.viewersUrl)
            .into(viewersUrl)
        viewersCount.text = news.viewersCount.toString()
        likesUrl.setOnClickListener { v ->
            if (news.liked == 1) {
                news.liked = 0
                news.likesCount = news.likesCount - 1
                Glide.with(likesUrl.context)
                    .load(R.drawable.likemin)
                    .into(likesUrl)
                likes.text = news.likesCount.toString()
                likes.setTextColor(Color.parseColor("#979899"))
                val toast = Toast.makeText(
                    v.context,
                    "Like removed!",
                    Toast.LENGTH_SHORT
                )
                toast.setGravity(Gravity.CENTER, 0, 0)
                toast.show()
                thisNews = news
            } else {
                news.liked = 1
                news.likesCount = news.likesCount + 1
                Glide.with(likesUrl.getContext())
                    .load(R.drawable.likebluemin)
                    .into(likesUrl)
                likes.text = news.likesCount.toString()
                likes.setTextColor(Color.parseColor("#006ba8"))
                val toast = Toast.makeText(
                    v.context,
                    "Liked!",
                    Toast.LENGTH_SHORT
                )
                toast.setGravity(Gravity.CENTER, 0, 0)
                toast.show()
                thisNews = news
            }
        }
    }
}